### PR TITLE
OCPEDGE-1273: chore: k8s 1.31 dependency bumps 

### PR DESCRIPTION
**Pull request by jakobmoellerdev**
Noteworthy changes:

- Role definitions in generated RBAC are now aggregated
- Some missing mandatory fields in the CRDs were patched in that were previously accidentally removed
- Documentation for the update process was added
- One of the mandatory replace statements was able to be removed
- The release includes the latest state of topolvm which has a filesystem expansion issue when using an older version of topolvm (see https://github.com/topolvm/topolvm/releases/tag/v0.33.0 for details)

### CODE REVIEW

File: cmd/vgmanager/vgmanager.go [hunk 1]
```diff
@@ -197,9 +196,6 @@ func run(cmd *cobra.Command, _ []string, opts *Options) error {
 			return fmt.Errorf("could not add topolvm metrics: %w", err)
 		}
 
-		if err := os.MkdirAll(topolvm.DeviceDirectory, 0755); err != nil {
```
Comment (suleymanakbas91) [inline cmd/vgmanager/vgmanager.go:200]: Why don't we need this anymore?

Comment (jakobmoellerdev) [reply cmd/vgmanager/vgmanager.go:200]: Because of https://github.com/topolvm/topolvm/releases/tag/v0.33.0 getting rid of the topolvm device directory

Comment (jakobmoellerdev) [reply cmd/vgmanager/vgmanager.go:200]: Concretely, this is the change: https://github.com/topolvm/topolvm/pull/930/files

**Pull request merged**